### ISSUE TITLE
🐞[Bug/Feature Request]: Logical Error on Food Donating Page and Proposal for Google Maps Integration

### ISSUE THREAD

**Issue by igopi77** [sanika391/SpoonShare#64]
**Description**
On the Food Donating page of the SpoonShare app, there is a logical mistake in the location input field. Users are currently required to type their location, which allows for the entry of invalid or nonsensical text (e.g., "dlkfjskdafhkadhf"). This results in inaccurate and unreliable location data.

**Comment**
What I See: Users can type anything into the location input field without any validation. This could lead to entries that are not actual locations, such as random strings or gibberish.

**Expected Behavior**
What I Expected to Happen: Users should be able to specify their current location accurately and intuitively. Ideally, the app should provide a map-based interface (such as Google Maps) for users to pinpoint their location. This would prevent the entry of invalid text and improve the accuracy of the location data.

**Video**

https://github.com/sanika391/SpoonShare/assets/123234840/2f7afe57-1d75-43fa-a978-0e52ad1c0213

I hope that I have a capability to solve this problem.

Thank you.

**Comment by igopi77** [comment]
Hello @sanika391 ,

My name is Gopinath, and I have a strong background in software development. If you're interested, you can view my work at the following repositories:

1. Helping Hand: [GitHub](https://github.com/igopi77/Helping-_Hands)
2. Stock Management: [GitHub](https://github.com/igopi77/Akshaya_Strore_V1)
3. Job Portal: [GitHub](https://github.com/MuthuvelA/JOBPORTAL_FRONTEND)

I am confident in my ability to make a significant contribution to your project. In particular, I believe I can successfully address the issue of the logical error in the location input and propose the integration of Google Maps for pinpointing locations within the SpoonShare project.

For more insight into my professional background and projects, please visit my profiles:

LinkedIn: [Profile](https://www.linkedin.com/in/gopinath-siva-191a27259/)
GitHub: [Profile](https://github.com/igopi77)

I am eager to explore the possibility of collaboration and engaging in meaningful discussions.

Warm regards,
Gopinath

**Comment by sanika391** [comment]
Certainly! Here’s a comment addressing the issue, assigning the work, and expressing enthusiasm to work with the contributor:

---

### Comment

**Issue Description:**
On the Food Donating page of the SpoonShare app, there is a logical mistake in the location input field. Currently, users are required to type their location manually. This setup allows for the entry of invalid or nonsensical text (e.g., "dlkfjskdafhkadhf"), leading to inaccurate and unreliable location data.

**What I See:**
Users can type anything into the location input field without any validation. This could result in entries that are not actual locations, such as random strings or gibberish.

**Expected Behavior:**
I expected users to be able to specify their current location accurately and intuitively. Ideally, the app should provide a map-based interface (such as Google Maps) for users to pinpoint their location. This would prevent the entry of invalid text and improve the accuracy of the location data.

**Video Reference:**
[Donate_Food.mp4](path_to_video)

---

Hi @igopi77,

Thank you for bringing this issue to our attention and providing a detailed explanation.

I'm pleased to assign you to this task. Your proposal to integrate a map-based interface like Google Maps for location input sounds like a perfect solution to ensure accurate and reliable data.

Feel free to start working on this enhancement. If you need any further information or assistance, don't hesitate to reach out. We are excited to have you on board and look forward to your contributions to the SpoonShare project.

Warm regards,

Sanika Chavan
Project Manager, SpoonShare
[https://www.linkedin.com/in/sanika-chavan-52457b236]

**Comment by igopi77** [comment]
Hello @sanika391 This issue has been merged. Can please close this issue.
